### PR TITLE
add anti-affinity for controller

### PR DESCRIPTION
**Pull request by curx**
Signed-off-by: Thorsten Schifferdecker <schifferdecker@b1-systems.de>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

add an anti-affinity spec to cis-nfs-controller

**Which issue(s) this PR fixes**:
see #173 

**Does this PR introduce a user-facing change?**:
```release-note
fix: add anti-affinity for the controller
```

### CODE REVIEW

File: charts/latest/csi-driver-nfs/templates/csi-nfs-controller.yaml [hunk 1]
```diff
@@ -15,6 +16,16 @@ spec:
 {{ include "nfs.labels" . | indent 6 }}
         app: csi-nfs-controller
     spec:
+      affinity:
```
Comment (andyzhangx) [inline charts/latest/csi-driver-nfs/templates/csi-nfs-controller.yaml:19]: should also update tgz file, check the guidance in ut failure, thanks.

Comment (curx) [reply charts/latest/csi-driver-nfs/templates/csi-nfs-controller.yaml:19]: thanks, I'll update it, as well

**Comment by andyzhangx** [comment]
this nfs driver is actually not using liveness probe correctly, otherwise the second controller pod on the same node would be in pending state since it's using hostNetwork, refer to:
https://github.com/kubernetes-csi/csi-driver-smb/blob/cd758b95117c24c51146fea0a536f04c91a7f5c3/deploy/csi-smb-controller.yaml#L70-L73

That's two issues 1) we should schedule controller pods on two different nodes 2) use probe correctly

**Comment by andyzhangx** [comment]
PR(https://github.com/kubernetes-csi/csi-driver-nfs/pull/175) fixed the issue, the second nfs controller pod would be in Pending status if there is only one agent node, here are the second pod events:
```
# k describe po csi-nfs-controller-5dfd6db785-slssf -n kube-system
Name:                 csi-nfs-controller-5dfd6db785-slssf
Namespace:            kube-system
Priority:             2000000000
Priority Class Name:  system-cluster-critical
Node:                 <none>
Labels:               app=csi-nfs-controller
                      pod-template-hash=5dfd6db785
Annotations:          <none>
Status:               Pending
IP:
IPs:                  <none>
Controlled By:        ReplicaSet/csi-nfs-controller-5dfd6db785
Containers:
  csi-provisioner:
    Image:      k8s.gcr.io/sig-storage/csi-provisioner:v2.1.0
    Port:       <none>
    Host Port:  <none>
    Args:
      -v=2
      --csi-address=$(ADDRESS)
      --leader-election
    Limits:
      cpu:     100m
      memory:  300Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      ADDRESS:                       /csi/csi.sock
      KUBERNETES_PORT_443_TCP_ADDR:  andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io
      KUBERNETES_PORT:               tcp://andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io:443
      KUBERNETES_PORT_443_TCP:       tcp://andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io:443
      KUBERNETES_SERVICE_HOST:       andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from csi-nfs-controller-sa-token-p59bw (ro)
  liveness-probe:
    Image:      k8s.gcr.io/sig-storage/livenessprobe:v2.1.0
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=/csi/csi.sock
      --probe-timeout=3s
      --health-port=29652
      --v=2
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      KUBERNETES_PORT_443_TCP_ADDR:  andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io
      KUBERNETES_PORT:               tcp://andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io:443
      KUBERNETES_PORT_443_TCP:       tcp://andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io:443
      KUBERNETES_SERVICE_HOST:       andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io
    Mounts:
      /csi from socket-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from csi-nfs-controller-sa-token-p59bw (ro)
  nfs:
    Image:      mcr.microsoft.com/k8s/csi/nfs-csi:latest
    Port:       29652/TCP
    Host Port:  29652/TCP
    Args:
      -v=5
      --nodeid=$(NODE_ID)
      --endpoint=$(CSI_ENDPOINT)
    Limits:
      cpu:     200m
      memory:  200Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Liveness:  http-get http://:healthz/healthz delay=30s timeout=10s period=30s #success=1 #failure=5
    Environment:
      NODE_ID:                        (v1:spec.nodeName)
      CSI_ENDPOINT:                  unix:///csi/csi.sock
      KUBERNETES_PORT_443_TCP_ADDR:  andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io
      KUBERNETES_PORT:               tcp://andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io:443
      KUBERNETES_PORT_443_TCP:       tcp://andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io:443
      KUBERNETES_SERVICE_HOST:       andy-aks11-andy-aks11962-b9d228-e3c955a0.hcp.eastus2euap.azmk8s.io
    Mounts:
      /csi from socket-dir (rw)
      /var/lib/kubelet/pods from pods-mount-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from csi-nfs-controller-sa-token-p59bw (ro)
Conditions:
  Type           Status
  PodScheduled   False
Volumes:
  pods-mount-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/pods
    HostPathType:  Directory
  socket-dir:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  csi-nfs-controller-sa-token-p59bw:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  csi-nfs-controller-sa-token-p59bw
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  kubernetes.io/os=linux
Tolerations:     node-role.kubernetes.io/master=true:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason            Age        From  Message
  ----     ------            ----       ----  -------
  Warning  FailedScheduling  <unknown>        0/2 nodes are available: 1 node(s) didn't have free ports for the requested pod ports, 1 node(s) were unschedulable.
```

**Pull request closed without merging**